sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosException.java

```diff
@@ -54,7 +54,7 @@ protected CosmosException(int statusCode, String message, Map<String, String> re
         super(message, cause);
         this.statusCode = statusCode;
         this.requestTimeline = RequestTimeline.empty();
-        this.responseHeaders = responseHeaders == null ? new HashMap<>() : new HashMap<>(responseHeaders);
+        this.responseHeaders = responseHeaders == null ? new HttpHeaders() : new HttpHeaders(responseHeaders);
```

Comment:
Can we construct HttpHeaders when we are creating Exception and then pass it along ? , it may avoid one extra step of Map creation

Reply:
Yup, that;s the next PR referred in the description.